src/ess/loki/workflow.py

```diff
@@ -1,24 +1,89 @@
 # This file is used by beamlime to create a workflow for the Loki instrument.
 # The function `live_workflow` is registered as the entry point for the workflow.
+import sciline
 import scipp as sc
-from beamlime import WorkflowResult
 from scippneutron.io.nexus.load_nexus import JSONGroup
 
+from ess.loki.general import (
+    get_monitor_data,
+    get_source_position,
+    monitor_to_tof,
+    patch_monitor_data,
+)
+from ess.loki.io import load_nexus_monitor, load_nexus_source
+from ess.sans.conversions import monitor_to_wavelength, sans_monitor
+from ess.sans.types import (  # FilePath,; Filename,; NeXusMonitorName,
+    Incident,
+    LoadedNeXusMonitor,
+    MonitorType,
+    RawSource,
+    RunType,
+    SampleRun,
+    Transmission,
+    WavelengthBins,
+    WavelengthMonitor,
+)
 
-def live_workflow(group: JSONGroup) -> WorkflowResult:
+
+class BinnedMonitor(
+    sciline.ScopeTwoParams[RunType, MonitorType, sc.DataArray], sc.DataArray
+):
+    ...
+
+
+def _bin_monitor_wavelength(
+    wavelength_bin: WavelengthBins, monitor: WavelengthMonitor[RunType, MonitorType]
+) -> BinnedMonitor[RunType, MonitorType]:
+    return monitor.bin(wavelength=wavelength_bin)
```

Comment:
Should we actually `hist`? Visualization should not have to do this itself, I think?